src/public/secure/files.ejs

```diff
@@ -1404,6 +1424,59 @@
                 dt.rows().deselect();
             }
         });
+
+        document.getElementById("bucketTools")?.addEventListener("click", async () => {
+            let bucketToken;
+            async function promoteYes() {
+                await setBucketType(bucketToken, "PREMIUM");
+            }
+
+            async function demoteYes() {
+                await setBucketType(bucketToken, "NORMAL");
+            }
+
+            async function generalNo() {
+                return;
+            }
+
+            async function yes() {
+                bucketToken = document.getElementById("bucketToken").value;
+                let resp;
+                try {
+                    site.loading(true);
+                    resp = await fetch(`${baseUrlSecureRest}/getBucketType/${bucketToken}`, {
```

Comment:
we have the bucket already. please add the type to the dto and remove this call.